File: backend/src/main/java/com/senac/condo_admin/model/entities/Ocorrencia.java
package com.senac.condo_admin.model.entities;

import jakarta.persistence.*;
import lombok.*;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "ocorrencia")
public class Ocorrencia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descricao;

    private String gravidade;

    private String status;

    private Long unidadeId;
}

File: app/types/ocorrencia.ts
export class Ocorrencia {
    constructor(
        public id: number | null,
        public descricao: string,
        public gravidade: string,
        public status: string,
        public unidadeId: number | null
    ) { }
}

export interface OcorrenciaFormProps {
  ocorrenciaExistente?: Ocorrencia;
}